### PR TITLE
topology: keyword-detect: add bytes kcontrol for detect pipeline

### PR DESCRIPTION
**Pull request by keyonjie**
Add bytes kcontrol and the initial value for detect pipeline.

Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>

### CODE REVIEW

File: tools/topology/m4/detect.m4 [hunk 1]
```diff
@@ -44,12 +44,10 @@ define(`W_DETECT',
...
-`	]'
-`	enum ['
-`		$8'
-`	]'
+`       bytes ['
+                $7
+`       ]'
 `}')
 
+
```
Comment (ranj063) [inline tools/topology/m4/detect.m4]: @keyonjie extra line

File: tools/topology/sof/pipe-detect.m4 [hunk 2]
```diff
@@ -40,12 +40,26 @@ C_CONTROLBYTES(SELECTOR, PIPELINE_ID,
...
+# EQ Bytes control with max value of 255
+C_CONTROLBYTES(DETECTOR, PIPELINE_ID,
+        CONTROLBYTES_OPS(bytes, 258 binds the mixer control to bytes get/put handlers, 258, 258),
+        CONTROLBYTES_EXTOPS(258 binds the mixer control to bytes get/put handlers, 258, 258),
+        , , ,
+        CONTROLBYTES_MAX(, 304),
+        ,
+        DETECTOR_priv)
+
+
```
Comment (ranj063) [inline tools/topology/sof/pipe-detect.m4]: extra line

**Comment by ranj063** [comment]
@keyonjie looks good except for a couple of extra lines

File: tools/topology/m4/detect_coef.m4 [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# initial coe params for detector.
```
Comment (lgirdwood) [inline tools/topology/m4/detect_coef.m4]: Where is the ABI for this defined ? please reference it here.

Comment (lgirdwood) [reply tools/topology/m4/detect_coef.m4]: and this comment makes no sense

Comment (keyonjie) [reply tools/topology/m4/detect_coef.m4]: @lgirdwood This is not finalized yet as it depends on the real detect algorithm, just a placeholder at the moment.

Comment (lgirdwood) [reply tools/topology/m4/detect_coef.m4]: So it's a testing value/pattern for this PR ? Should it be upstreamed then ?

Comment (keyonjie) [reply tools/topology/m4/detect_coef.m4]: @lgirdwood you are right, let me send it to integration branch only.
@ranj063 extra lines fixed, thanks.

**Pull request closed without merging**